python/gui/qgsadvanceddigitizingdockwidget.sip

```diff
@@ -1,337 +1,377 @@
-     * @param canvas The map canvas on which the widget operates
-     * @param parent The parent
-     */
-    explicit QgsAdvancedDigitizingDockWidget( QgsMapCanvas* canvas, QWidget *parent = 0 );
-
-    /**
-     * Disables the CAD tools when hiding the dock
-     */
-    void hideEvent( QHideEvent* );
-
-    /**
-     * Will react on a canvas press event
-     *
-     * @param e A mouse event (may be modified)
-     * @return  If the event is hidden (construction mode hides events from the maptool)
-     */
-    bool canvasPressEvent( QgsMapMouseEvent* e );
-    /**
-     * Will react on a canvas release event
-     *
-     * @param e A mouse event (may be modified)
-     * @param mode determines if the dock has to record one, two or many points.
-     * @return  If the event is hidden (construction mode hides events from the maptool)
-     */
-    bool canvasReleaseEvent( QgsMapMouseEvent* e, AdvancedDigitizingMode mode );
-    /**
-     * Will react on a canvas move event
-     *
-     * @param e A mouse event (may be modified)
-     * @return  If the event is hidden (construction mode hides events from the maptool)
-     */
-    bool canvasMoveEvent( QgsMapMouseEvent* e );
-    /**
-     * Filter key events to e.g. toggle construction mode or adapt constraints
-     *
-     * @param e A mouse event (may be modified)
-     * @return  If the event is hidden (construction mode hides events from the maptool)
-     */
+    static bool lineCircleIntersection( const QgsPoint &center, const double radius, const QList<QgsPoint> &segment, QgsPoint &intersection );
+%Docstring
+.. note::
+
+   from the two solutions, the intersection will be set to the closest point
+ :rtype: bool
+%End
+
+    explicit QgsAdvancedDigitizingDockWidget( QgsMapCanvas *canvas, QWidget *parent = 0 );
+%Docstring
+ Create an advanced digitizing dock widget
+ \param canvas The map canvas on which the widget operates
+ \param parent The parent
+%End
+
+    virtual void hideEvent( QHideEvent * );
+
+%Docstring
+ Disables the CAD tools when hiding the dock
+%End
+
+    bool canvasPressEvent( QgsMapMouseEvent *e );
+%Docstring
+ Will react on a canvas press event
+
+ \param e A mouse event (may be modified)
+ :return:  If the event is hidden (construction mode hides events from the maptool)
+ :rtype: bool
+%End
+
+    bool canvasReleaseEvent( QgsMapMouseEvent *e, AdvancedDigitizingMode mode );
+%Docstring
+ Will react on a canvas release event
+
+ \param e A mouse event (may be modified)
+ \param mode determines if the dock has to record one, two or many points.
+ :return:  If the event is hidden (construction mode hides events from the maptool)
+ :rtype: bool
+%End
+
+    bool canvasMoveEvent( QgsMapMouseEvent *e );
+%Docstring
+ Will react on a canvas move event
+
+ \param e A mouse event (may be modified)
+ :return:  If the event is hidden (construction mode hides events from the maptool)
+ :rtype: bool
+%End
+
     bool canvasKeyPressEventFilter( QKeyEvent *e );
```

Comment:
this should be put in the private section.

just add this in the private section of the header:
    #ifdef SIP_RUN
       bool canvasKeyPressEventFilter( QKeyEvent *e );
    #endif